src/include/buffer.h

```diff
@@ -1132,6 +1132,8 @@ struct error_code;
     void append(ptr&& bp);
     void append(const ptr& bp, unsigned off, unsigned len);
     void append(const list& bl);
+    /// append data from the stream line by line with '\n' inserted between,
+    /// and at the end append '\n' even the stream does not end with EOL.
```

Comment:
it is surprising. as it could append something that does not exist in `in`. but might want to note '\n' is not inserted after an empty line? so probably something like

> append data from the stream line by line and add '\n' after each appended lines.
> if a line is empty, '\n' is not added.

Reply:
OK. I mentioned about non-empty line and added an example from the unit test.